pkg/types/types.go

```diff
@@ -127,6 +129,9 @@ type BasicRuntimeMetadata struct {
 	// response with multiple containers, ContainerName contains only the first element.
 	ContainerName string `json:"containerName,omitempty" column:"containerName,template:container"`
 
+	// ContainerPID is the process ID of the first process in the container.
+	ContainerPID uint32 `json:"containerPid,omitempty" column:"containerPid,template:pid,hide"`
+
```

Comment:
We already have a PID field in the container structure. I think we need to remove that to avoid duplication. This is the reason why this is not working as well for built-in gadgets, this field is never been set:

```diff
$ git diff
diff --git a/pkg/container-collection/container-collection.go b/pkg/container-collection/container-collection.go
index 2d9e87fa4..b4b7c2f59 100644
--- a/pkg/container-collection/container-collection.go
+++ b/pkg/container-collection/container-collection.go
@@ -210,6 +210,7 @@ func (cc *ContainerCollection) AddContainer(container *Container) {
                }
        }

+       container.Runtime.ContainerPID = container.Pid
        _, loaded := cc.containers.LoadOrStore(container.Runtime.ContainerID, container)
        if loaded {
                return
```

Reply:
Thanks, it works with your patch. But I don't see any duplication in the yaml output:

```bash
$ sudo ig trace exec -o yaml -c test
```
```yaml
---
args:
- /usr/bin/sh
- -c
- cat /dev/null
comm: sh
gid: 0
group: root
k8s:
  owner: {}
loginuid: 4294967295
mountnsid: 4026532891
pcomm: containerd-shim
pid: 45296
ppid: 45273
pupperlayer: false
runtime:
  containerId: 7ba96e85c2df96108036c73379b513d2797b2ea43f2fbdc0888981cf55708353
  containerImageName: ubuntu
  containerName: test
  containerPid: 45296
  containerStartedAt: 1729592047995532764
  runtimeName: docker
sessionid: 4294967295
timestamp: 1729592048136816332
type: normal
uid: 0
upperlayer: false
user: root

---
args:
- /usr/bin/cat
- /dev/null
comm: cat
gid: 0
group: root
k8s:
  owner: {}
loginuid: 4294967295
mountnsid: 4026532891
pcomm: sh
pid: 45319
ppid: 45296
pupperlayer: false
runtime:
  containerId: 7ba96e85c2df96108036c73379b513d2797b2ea43f2fbdc0888981cf55708353
  containerImageName: ubuntu
  containerName: test
  containerPid: 45296
  containerStartedAt: 1729592047995532764
  runtimeName: docker
sessionid: 4294967295
timestamp: 1729592048138417372
type: normal
uid: 0
upperlayer: false
user: root
```

Reply:
I wasn't clear enough, sorry. I meant that the container.Pid field

https://github.com/inspektor-gadget/inspektor-gadget/blob/67f2288c9d774037ece7bcb25eacb7a9ec7ed618/pkg/container-collection/containers.go#L41-L42

and the container.Runtime.ContainerPID field represents the same information:

https://github.com/inspektor-gadget/inspektor-gadget/blob/67f2288c9d774037ece7bcb25eacb7a9ec7ed618/pkg/types/types.go#L132-L133

So probably we want to remove the former and leave only the later.

Reply:
Thanks for the clarification. I updated the PR.